sql/mob_groups.sql

```diff
@@ -5298,7 +5298,7 @@ INSERT INTO `mob_groups` VALUES (23,6546,88,'Enchanted_Bones_blm',330,1,677,0,0,
 INSERT INTO `mob_groups` VALUES (24,6419,88,'Rock_Eater',330,0,1725,0,0,32,35,0);
 INSERT INTO `mob_groups` VALUES (25,1623,88,'Gloomanita',0,32,1006,0,0,42,53,0);
 INSERT INTO `mob_groups` VALUES (26,6464,88,'Fledermaus',330,2,82,0,0,45,47,0);
-INSERT INTO `mob_groups` VALUES (27,5870,88,'Olgoi-Khorkhoi',0,128,3285,4500,0,53,54,0);
```

Comment:
it was likely intended to use the zone lua's onZoneItitialize to set up the 1st spawn period, as the spawn time in the group is also zero and it shouldn't already be spawned at server start.

This would be why someone set it to scripted.

Reply:
That'll call for a rewrite of many other NMs that have this coded this way. Can't say all but some random timed spawns that I looked at are coded as spawntype 0 with respawn timer onMobDespawn.

Reply:
uh, no. we've ton set up exactly as I just described. some others have a static initial spawn period that then gets overwritten. Very few if any are zero time and zero spawn type and then also set a random time on death.

Reply:
Does the slot to the left of the spawntype slot, start running on server initialize? It's supposedly the respawn slot. How would I do this onInitialize? Is there a particular mob/nm that has code like what you're asking?

Reply:
there are a **lot** of existing examples:
- https://github.com/LandSandBoat/server/blob/base/scripts/zones/Beaucedine_Glacier/Zone.lua#L10
- https://github.com/LandSandBoat/server/blob/base/scripts/zones/Bhaflau_Thickets/Zone.lua#L13
- https://github.com/LandSandBoat/server/blob/base/scripts/zones/Caedarva_Mire/Zone.lua#L9

Reply:
Ok, so call it in the zone.lua itself for first spawn. Does that `GetMobByID(ID.mob.AYNU_KAYSEY):setRespawnTime(math.random(900, 10800))` spawn it when the zone is initializes? To me, that just looks to reset the respawn timer. Olgoi doesn't have an entry in nm_spawn_points.sql so having it updatenmspawnpoints won't work.

Reply:
it sets the time period. if it didn't have one. it will after that line executes. if it did have a timer that wasn't hit yet, it'll replace it.

Reply:
this also means you can set spawn type 0 and give it a timer in the group, and this will replace it. I just don't want it to be 0 AND ALSO have a zero spawn time. make sense?

Reply:
The respawn time is already coded within the onMobDespawn in it's lua. That's why I opted to use 0.

Reply:
> The respawn time is already coded within the onMobDespawn in it's lua. That's why I opted to use 0.

you aren't understanding there - the script doesn't set the initial spawn timing, the sql does. these are not interchangeable.

Reply:
at the time the mobs onMobDeath is read, it was ALREADY initialized. if that weren't they case we'd never need to use the zone to set them up.